### ISSUE TITLE
Bug - Switch - use :focus-visible rather than :focus for outline

### ISSUE THREAD

**Issue by keithjgrant** [patternfly/patternfly#5481]
**Describe the problem**
The focus ring on the Switch component shows up on click

**How do you reproduce the problem?**
Click a Switch to check/uncheck it

**Expected behavior**
Ideally the focus ring would only show with keyboard interaction

**Is this issue blocking you?**
No

**Screenshots**
![Screenshot 2023-04-13 at 2 50 08 PM](https://user-images.githubusercontent.com/410794/231892365-ef285c44-59f6-44f5-97f1-cba8f173d6ba.png)

**What is your product and what release date are you targeting?**
Ansible Controller 2.5

**Any other information?**

I propose changing the selector `.pf-c-switch__input:focus ~ .pf-c-switch__toggle` in [switch.scss](https://github.com/patternfly/patternfly/blob/v5/src/patternfly/components/Switch/switch.scss#L74) to `.pf-c-switch__input:focus-visible ~ .pf-c-switch__toggle` unless there's an older browser version we're trying to support that prevents that

**Comment by nicolethoen** [comment]
We'd need to double check that removing focus styles for mouse users does not violate a11y standards.
